### PR TITLE
Web & Device Extension by InfinityLoop

### PR DESCRIPTION
**Pull request by InfinityLoop1**
<h1>Web & Device</h1>
<h2>Created by InfinityLoop</h2>

Web & Device is an extension created by InfinityLoop (InfinityLoopGaming on Scratch), to interact and get data from websites, your device, and your network.

Features include:
- Change & get page title
- Get network stats
- Get browser, OS, and memory stats
- Toggle & get fullscreen
- Open and redirect tabs, get current tab, get data from URL
- Get screen width/height
- Get and set clipboard (not included in picture because it was added later)
- Get battery stats and events (not included in picture because it was added later)

By the way, I have gotten comments that all of these blocks exist in other extensions. This extension is meant to combine very similar extensions, like FETCH, Navigator Info, and Network. Because it would be easier to have these all in one, instead of having to install 2 or 3 extensions to do what you want, or having to find those extensions in a long list.

><b>Note</b>
>I wasn't aware of the “no combination extensions” rule before I had started making this extension, because this is my first extension.

### CODE REVIEW

**Comment by CubesterYT** [comment]
Hello @InfinityLoopGames, Currently, the blocks from this extension are present in these other extensions:

`(get data from url ())` Exists in the fetch extension.

`<connected to internet>`, `(network type)`, `(network generation)`, `(browser)`, `(current url)`, `open url () in new tab`, `redirect to url ()` Exists in the Networking Extension.

`(operating system)` and `(device memory)` Exist in Navigator Info Extension.

`(page title)`, `set page title to ()`, `<is page fullscreen>`, `enter fullscreen`, `exit fullscreen`, `(screen width)`, `(screen height)` Exist in Window Controls Extension.

**Comment by InfinityLoopGames** [comment]
@CubesterYT
Yes, this is kind of meant to combine similar extensions, they are also changed a bit.
I credited them in the code.

**Comment by CubesterYT** [comment]
Alright, but I'll want to let you know that this extension may not be merged because
 a. General Utility Extensions aren't allowed
 b. Combination Extensions don't usually fly
 c. Personal type extensions should just stay personal, instead of submitting a PR for public use.

These are part of the guidelines, but you are still allowed to keep this up till GarboMuffin reviews this.

**Comment by InfinityLoopGames** [comment]
I understand how it is the first two things you said, but how is this a personal type extension?

**Comment by CubesterYT** [comment]
> I understand how it is the first two things you said, but how is this a personal type extension?

@InfinityLoopGames, I didn't mean that this was a personal one, I'm just stating general rejected extensions.

**Comment by InfinityLoopGames** [comment]
@CubesterYT oh ok

**Comment by InfinityLoopGames** [comment]
Oh ok, I'll do that next time I make an extension.

**Review by LilyMakesThings — COMMENTED** [review]
I was going to review this PR in full, but as has been mentioned, a lot of this has been copy-pasted from other extensions, so I'm not going to go through the effort of reviewing the code a second time.

If you had read the contributing guidelines before submitting an extension, you would be familiar with the rule on extensions that are essentially the same as another extension (as well as "utility" extensions, that don't do a single thing).

File: extensions/InfinityLoopGaming/WebandDevice.js [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+          },
+          {
+            opcode: "currenturl",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("current link"),
+          },
+          {
+            opcode: "getdatafromurl",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "get data from url [url]",
```
[the remark this answers is not in the document]
Comment (InfinityLoopGames) [reply extensions/InfinityLoopGaming/WebandDevice.js:98]: Isn't that the only word in Fetch?

**Comment by Martinelplayz** [comment]
> I understand how it is the first two things you said, but how is this a personal type extension?

i mean, there is always sharkpool's gallery.

File: extensions/InfinityLoopGaming/WebandDevice.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (CST1229) [reply extensions/InfinityLoopGaming/WebandDevice.js:98]: In fetch there isn't any other noun the block could use (e.g naming the block `([url])` would be weird), and it's also GET for GET requests.

[the remark this answers is not in the document]
Comment (InfinityLoopGames) [reply extensions/InfinityLoopGaming/WebandDevice.js:98]: So what should I use instead?

[the remark this answers is not in the document]
Comment (CubesterYT) [reply extensions/InfinityLoopGaming/WebandDevice.js:98]: Use `fetch`

**Comment by InfinityLoop1** [comment]
Fixed some issues with the extension ID, should work correctly now when ran as a file.

**Comment by GarboMuffin** [comment]
i'm not going to say no at this point, but as these all kind of exist elsewhere, it'll probably be a while

i know these blocks are currently very disorganized; centralizing some of the random ones into an extension that people will actually be able to find is not a bad idea

**Comment by InfinityLoop1** [comment]
Added block icons

**Pull request closed without merging**